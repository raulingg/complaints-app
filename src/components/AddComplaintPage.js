import React, { Component } from 'react';
import { connect } from 'react-redux';
import ComplaintFormComponent from './ComplaintForm';
import { startAddComplaint as startAddComplaintAction } from '../actions/complaints';

export class AddComplaintPage extends Component {
  onSubmit = complaint => {
    const { history, startAddComplaint } = this.props;
    startAddComplaint(complaint);
    history.push('/');
  };

  render() {
    return (
      <div>
        <h1>Reporta tu denuncia </h1>
        <ComplaintFormComponent onSubmit={this.onSubmit} />
      </div>
    );
  }
}
const mapStateToProps = (state, props) => ({
  complaint: state.complaints.find(complaint => complaint.id === props.match.params.id),
});

const mapDispatchToProps = dispatch => ({
  startAddComplaint: complaint => dispatch(startAddComplaintAction(complaint)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddComplaintPage);
