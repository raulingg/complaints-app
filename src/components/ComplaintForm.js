import React, { Component } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { startAddComplaint as startAddComplaintAction } from '../actions/complaints';

export class ComplaintForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: props.complaint ? props.complaint.title : '',
      content: props.complaint ? props.complaint.content : '',
      reportTo: props.complaint ? props.complaint.reportTo : '',
      happenedAt: props.complaint ? moment(props.complaint.happenedAt) : moment(),
    };
  }

  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };

  onContentChange = e => {
    const content = e.target.value;
    this.setState(() => ({ content }));
  };

  onReportToChange = e => {
    const reportTo = e.target.value;
    this.setState(() => ({ reportTo }));
  };

  onHappenedAtChange = e => {
    const happenedAt = e.target.value;
    this.setState(() => ({ happenedAt }));
  };

  onSubmit = e => {
    e.preventDefault();
    const { title, content, happenedAt, reportTo } = this.state;
    const { startAddComplaint, history } = this.props;
    startAddComplaint({
      title,
      content,
      happenedAt,
      reportTo,
    });
    history.push('/');
  };

  render() {
    const { title, content, happenedAt, reportTo } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <div>
          <input type="text" name="title" value={title} onChange={this.onTitleChange} />
        </div>
        <div>
          <textarea
            name="content"
            maxLength="2000"
            rows="10"
            value={content}
            onChange={this.onContentChange}
          />
        </div>
        <div>
          <input
            type="date"
            name="happenedAt"
            value={happenedAt}
            onChange={this.onHappenedAtChange}
          />
        </div>
        <div>
          <input type="text" name="reportTo" value={reportTo} onChange={this.onReportToChange} />
        </div>
        <div>
          <button className="button">Guardar cambios</button>
        </div>
      </form>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddComplaint: complaint => dispatch(startAddComplaintAction(complaint)),
});

export default connect(
  undefined,
  mapDispatchToProps
)(ComplaintForm);
