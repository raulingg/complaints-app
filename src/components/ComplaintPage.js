import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

export const ComplaintPage = ({ complaint }) => (
  <div>
    <div>
      <img src={complaint.user.photoURL} alt="" />
      <span>{complaint.user.displayName}</span>
    </div>
    <div>
      <h1>{complaint.title}</h1>
      <p>
        <span>Denunciado el {moment(complaint.happenedAt).format()}</span>
      </p>
      <p>
        Denuncia a: <span>{complaint.reportTo}</span>
      </p>
      <div dangerouslySetInnerHTML={{ __html: complaint.content }} />
    </div>
  </div>
);

const mapStateToProps = (state, props) => ({
  complaint: state.complaints.find(complaint => String(complaint.id) === props.match.params.id),
});

export default connect(mapStateToProps)(ComplaintPage);
