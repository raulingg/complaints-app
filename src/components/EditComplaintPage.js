import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import ComplaintFormComponent from './ComplaintForm';
import { startEditComplaint as startEditComplaintAction } from '../actions/complaints';
import withAuthorization from './auth/withAuthorization';

export class EditComplaintPage extends Component {
  onSubmit = updates => {
    const {
      history,
      startEditComplaint,
      complaint: { id },
    } = this.props;
    startEditComplaint(id, updates);
    history.push('/');
  };

  render() {
    const { complaint } = this.props;
    return (
      <div>
        <h1>Editar denuncia </h1>
        <ComplaintFormComponent complaint={complaint} onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  complaint: state.complaints.find(complaint => String(complaint.id) === props.match.params.id),
});

const mapDispatchToProps = dispatch => ({
  startEditComplaint: (id, complaint) => dispatch(startEditComplaintAction(id, complaint)),
});

export default compose(
  withRouter,
  withAuthorization,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(EditComplaintPage);
