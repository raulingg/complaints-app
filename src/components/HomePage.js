import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Button from '../styles/components/Buttons';
import { startSetComplaints as startSetComplaintsAction } from '../actions/complaints';
import LoadingPage from './LoadingPage';

export class HomePage extends Component {
  state = {
    isLoading: true,
  };

  async componentDidMount() {
    const { startSetComplaints } = this.props;
    await startSetComplaints();
    this.setState({ isLoading: false });
  }

  render() {
    const { complaints } = this.props;
    const { isLoading } = this.state;

    return isLoading ? (
      <LoadingPage />
    ) : (
      <div>
        <div>
          <div>Denuncia</div>
          <div>Tipo</div>
          <div>Fecha</div>
        </div>
        <div>
          <Button as={Link} to="/add">
            Denunciar
          </Button>
          {complaints.length === 0 ? (
            <div>
              <span>No encontramos ninguna denuncia.</span>
            </div>
          ) : (
            complaints.map(complaint => (
              <div key={complaint.id}>
                <p>
                  <Link to={`show/${complaint.id}`}>{complaint.title}</Link>
                </p>
              </div>
            ))
          )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  complaints: state.complaints,
});

const mapDispatchToProps = dispatch => ({
  startSetComplaints: () => dispatch(startSetComplaintsAction()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomePage);
