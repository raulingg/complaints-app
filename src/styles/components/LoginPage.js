import styled from 'styled-components';
import { sizes } from '../settings';

const Container = styled.div`
  align-items: center;
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
`;

const Box = styled.div`
  background: rgba(255, 255, 255, 0.85);
  border-radius: 3px;
  padding: ${sizes.l} ${sizes.m};
  text-align: center;
  width: 25rem;
`;

const Title = styled.h1`
  margin: 0 0 ${sizes.m} 0;
  line-height: 1;
`;
export { Box, Container, Title };
