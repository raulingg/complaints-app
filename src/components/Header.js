import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startLogout as startLogoutAction } from '../actions/auth';
import * as HeaderStyles from '../styles/components/Header';
import ContentContainer from '../styles/components/ContentContainer';
import { ButtonLink } from '../styles/components/Buttons';

export const Header = ({ isAuthenticated, startLogout }) => (
  <HeaderStyles.Header>
    <ContentContainer>
      <HeaderStyles.Content>
        <HeaderStyles.Title to="/">
          <h1>Denuncias</h1>
        </HeaderStyles.Title>
        {isAuthenticated ? (
          <ButtonLink onClick={startLogout}>Salir</ButtonLink>
        ) : (
          <ButtonLink as={Link} to="/login">
            Iniciar sesi√≥n
          </ButtonLink>
        )}
      </HeaderStyles.Content>
    </ContentContainer>
  </HeaderStyles.Header>
);

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.user,
});

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogoutAction()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
