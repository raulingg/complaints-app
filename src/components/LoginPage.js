import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin as startLoginAction } from '../actions/auth';
import * as LoginPageStyles from '../styles/components/LoginPage';
import Button from '../styles/components/Buttons';

export const LoginPage = ({ history, startLogin }) => {
  const onClick = async () => {
    await startLogin();
    history.push('/');
  };

  return (
    <LoginPageStyles.Container>
      <LoginPageStyles.Box>
        <LoginPageStyles.Title>Denuncia Per√∫ App</LoginPageStyles.Title>
        <Button onClick={onClick}>Login con Google</Button>
        <p>Haz tu denuncia, no calles!.</p>
      </LoginPageStyles.Box>
    </LoginPageStyles.Container>
  );
};

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLoginAction()),
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(LoginPage)
);
