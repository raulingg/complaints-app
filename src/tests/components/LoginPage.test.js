import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';
import Button from '../../styles/components/Buttons';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', done => {
  const historySpy = {
    push: jest.fn(),
  };
  const startLoginSpy = jest.fn(() => Promise.resolve('done'));

  const wrapper = shallow(<LoginPage startLogin={startLoginSpy} history={historySpy} />);
  wrapper.find(Button).simulate('click');
  expect(startLoginSpy).toHaveBeenCalledTimes(1);
  // expect(historySpy.push).toHaveBeenCalledTimes(1);
  done();
});
