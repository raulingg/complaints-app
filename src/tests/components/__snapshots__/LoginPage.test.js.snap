// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<ForwardRef>
  <ForwardRef>
    <ForwardRef>
      Denuncia Per√∫ App
    </ForwardRef>
    <ForwardRef
      onClick={[Function]}
    >
      Login con Google
    </ForwardRef>
    <p>
      Haz tu denuncia, no calles!.
    </p>
  </ForwardRef>
</ForwardRef>
`;
